ArchivesSpace: Automate matching GUI

Hwesta opened the pull request:
Automatically parse an `archivesspacesids.csv` and populate the matching GUI for archivesspace.
- Do not automatically open the matching GUI
- Add DIP generation helper to Prepare DIP that parses `archivesspaceids.csv` for `filename,reference ID`, queries ArchivesSpace for the resource ID associated with the reference identifier, and populates ArchivesSpaceDIPObjectResourcePairing

Hwesta committed:
ArchivesSpace: Do not open matching GUI with JS

Hwesta committed:
DIP: Add DIP generation helper skeleton

Helper to generate information needed by the DIP that is only available in
the SIP. E.g. parsing metadata files.

Hwesta committed:
ArchivesSpace: Parse archivesspaceids.csv for filename->resource ID

Hwesta committed:
ArchivesSpace: Minor fixes

Hwesta committed:
ArchivesSpace: Add query by field function

Add function to return a resource based on an exact query by a specific field.

Hwesta committed:
Add helper to collect all metadata files

Helper searches transfer and SIP metadata directories for all copies of a
file with the provided filename.

mistydemeo commented on src/MCPClient/lib/clientScripts/dip_generation_helper.py:
```diff
@@ -0,0 +1,132 @@
+
+    rc = 0
+
+    rc = rc or parse_archivesspace_ids(args.sipPath, args.sipUUID)
```
Since this exits 0 on success, can you just `sys.exit(parse_archivesspace_ids(args.sipPath, args.sipUUID))`?

Hwesta replied:
I was thinking this script would be used for multiple DIP helpers, and the script should return failure if any one of them failed, so the RC should be the or'd results of all the helpers. Does this seem reasonable?

mistydemeo replied:
That makes sense. Maybe leave a comment to that effect (since it's not clear when there aren't yet multiple values to `or`), or do that when there is more than one exit code?

Hwesta replied:
Updated

mistydemeo commented on src/dashboard/src/media/js/ingest.js:
```diff
@@ -331,13 +331,6 @@ $(function()
             window.location.href = '/ingest/' + this.model.sip.get('uuid') + '/upload/atk/';
           }
 
-          // redirect to object/resource mapping pages
```
Will the user need to manually click through in this case?

Hwesta replied:
Yes.  Since this PR adds automating the matching GUI (for ArchivesSpace), I removed starting up the matching GUI in JS. I'd rather it checked if it was necessary first, but that's non-trivial.  Open to discussion on that one.

mistydemeo replied:
It'd probably be good to ask Justin/Sarah about this, but given the circumstances this is probably the best compromise. Maybe we could make the way you open the matching GUI manually more obvious to compensate?

mistydemeo commented:
:+1: Tests are nice to see too :D

Hwesta committed:
ArchivesSpace: Provide access file size & format info

refs #8696

When uploading to ArchivesSpace, also upload the access file's size and
format name.  file_format_name is a restricted vocabulary and should be
configured in ArchivesSpace first so upload succeeds.

Hwesta committed:
ArchivesSpace: add subjects & agents to children

refs #8697

Hwesta commented:
Note: this is in the process of getting updated other ArchivesSpace enhancements (tickets #8697 and #8696 in Redmine)

Hwesta committed:
Dashboard: AS/ATK cleanup

Remove unused function, use messages for form confirmation, update ATK config
saving to match AS, format template.

Hwesta commented:
Now based on top of #261 

Added 0d065ecd2af7bb1c575ff13bd99246ceccdc9177
